pwiz_tools/Skyline/Model/AlphaPeptDeep/AlphaPeptDeepLibraryBuilder.cs

```diff
@@ -0,0 +1,405 @@
+using pwiz.Skyline.Model.Lib;
+using pwiz.Skyline.Model.Tools;
+using pwiz.Skyline.Util.Extensions;
+
+namespace pwiz.Skyline.Model.AlphaPeptDeep
+{
+    public class AlphapeptdeepLibraryBuilder : IiRTCapableLibraryBuilder
+    {
+        private const string ALPHAPEPTDEEP = @"alphapeptdeep";
+        private const string BLIB_BUILD = "BlibBuild";
+        private const string CHARGE = @"charge";
+        private const string CMD_FLOW_COMMAND = @"cmd-flow";
+        private const string EXPORT_SETTINGS_COMMAND = @"export-settings";
+        private const string EXT_TSV = TextUtil.EXT_TSV;
+        private const string INPUT = @"input";
+        private const string LEFT_PARENTHESIS = TextUtil.LEFT_PARENTHESIS;
+        private const string LEFT_SQUARE_BRACKET = TextUtil.LEFT_SQUARE_BRACKET;
+        private const string LIBRARY_COMMAND = @"library";
+        private const string MOD_SITES = @"mod_sites";
+        private const string MODIFIED_PEPTIDE = "ModifiedPeptide";
+        private const string MODS = @"mods";
+        private const string OUTPUT = @"output";
+        private const string OUTPUT_MODELS = @"output_models";
+        private const string OUTPUT_SPECTRAL_LIB_FILE_NAME = @"predict.speclib.tsv";
+        private const string OUTPUT_SPECTRAL_LIBS = @"output_spectral_libs";
+        private const string PEPTDEEP_EXECUTABLE = "peptdeep.exe";
+        private const string RIGHT_PARENTHESIS = TextUtil.RIGHT_PARENTHESIS;
+        private const string RIGHT_SQUARE_BRACKET = TextUtil.RIGHT_SQUARE_BRACKET;
+        private const string SEMICOLON = TextUtil.SEMICOLON;
+        private const string SEQUENCE = @"sequence";
+        private const string SETTINGS_FILE_NAME = @"settings.yaml";
+        private const string SPACE = TextUtil.SPACE;
+        private const string TAB = "\t";
+        private const string TRANSFORMED_OUTPUT_SPECTRAL_LIB_FILE_NAME = @"predict_transformed.speclib.tsv";
+        private const string UNDERSCORE = TextUtil.UNDERSCORE;
+
+        /// <summary>
+        /// key: unimod ID, value: modification name supported by Alphapeptdeep
+        /// </summary>
+        private static readonly Dictionary<int, string> AlphapeptdeepModificationName = new Dictionary<int, string>()
+        {
+            {4, @"Carbamidomethyl@C"},
+            {21, @"Phospho@S"},
+            {35, @"Oxidation@M"},
+        };
+        private static readonly IEnumerable<string> PrecursorTableColumnNames = new[] { SEQUENCE, MODS, MOD_SITES, CHARGE };
+
+        private string PythonVirtualEnvironmentScriptsDir { get; }
+        private string PeptdeepExecutablePath => Path.Combine(PythonVirtualEnvironmentScriptsDir, PEPTDEEP_EXECUTABLE);
+        private string RootDir => Path.Combine(ToolDescriptionHelpers.GetToolsDirectory(), ALPHAPEPTDEEP);
+        private string SettingsFilePath => Path.Combine(RootDir, SETTINGS_FILE_NAME);
+        private string InputFileName => INPUT + UNDERSCORE +Document.DocumentHash +EXT_TSV;
```

Comment:
"Document.DocumentHash" might contain the character "/" which would cause problems here.

You could use "PathEx.ReplaceInvalidFilenameCharacters" to get rid of that.
It might make more sense to put all of the temporary files into a subfolder of the .blib file that is being created. That is, if the user is building "foo.blib" then all of the temporary files go into "foo.blib.temp".
We know that the user has write permissions on the folder where the .blib file is being created.
I am not sure whether we are guaranteed that the user will be able to write to the tools directory.